### DISCUSSION TITLE
Link entries in another language

### DISCUSSION THREAD

**Discussion by mihob** [craftcms/cms#11111] [Q&A] [answered]
Is it somehow possible to select a language version of an entry via an entry field?

In the field configuration it is said that exactly this information is not stored in the relation. However, we would like to be able to link language versions of entries without having to store a fixed url. 

Is there any solution for this?

**Comment by brandonkelly** [comment, marked as answer]
No, sorry. Relations only store the target element ID, not its site. You could keep track of the target site in a separate dropdown field, though. It could even be built dynamically using [LJ Dynamic Fields](https://plugins.craftcms.com/craft-dynamic-fields?cmsConstraint=%5E4.0), with the following template:

```twig
{% set options = craft.app.sites.allSites|map(s => {label: s.name, value: s.id}) %}
{% set options = options|unshift({label: 'Any', value: ''}) %}
{{ options|json_encode }}
```

In your templates, you could then check whether a site was selected, and if so, update the relational query based on it:

```twig
{% set query = entry.myRelationalField %}

{% if entry.mySiteField %}
    {% do query.siteId(entry.mySiteField)
{% endif %}

{% set elements = query.all() %}
```
